File: leetCode/linovce/lc_448.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        Arrays.sort(nums);
        List<Integer> list = new ArrayList<Integer>();
        int t=1;
        for(int i=0;i<nums.length&&t<=nums.length;){
            if(nums[i]==t){
                i++;
                t++;
            }else if(nums[i]<t){
                i++;
            }else{
                list.add(t);
                t++;
            }
        }
        while(t<=nums.length){
            list.add(t);
            t++;
        }
        return list;
    }
}

File: leetCode/fukun/401-500/lc_448.js
var findDisappearedNumbers = function (nums) {
    for (let i = 0; i < nums.length; i++) nums[Math.abs(nums[i]) - 1] = nums[Math.abs(nums[i]) - 1] > 0 ? -nums[Math.abs(nums[i]) - 1] : nums[Math.abs(nums[i]) - 1];
    return nums.reduce((a, b, i) => {
        if (b > 0) a.push(i + 1);
        return a;
    }, []);
};
console.log(findDisappearedNumbers([4, 3, 2, 7, 8, 2, 3, 1]))